### PR TITLE
[CPU] TopK node migration on nGraph.

### PR DESCRIPTION
**Pull request by nshchego**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/topk.cpp [hunk 1]
```diff
@@ -77,33 +82,27 @@ class TopKImpl: public ExtLayerBase {
...
             dim = static_cast<int>(src_dims[axis]);
             before_num = count(src_dims, 0, axis);
 
-            if (layer->outData.size() == 1) {
-                addConfig(layer, { DataConfigurator(ConfLayout::PLN, Precision::FP32), DataConfigurator(ConfLayout::PLN, Precision::I32) },
-                    { DataConfigurator(ConfLayout::PLN) });
+            if (topK1Op->get_output_size() == 1) {
+                addConfig(op, {{TensorDescCreatorTypes::ncsp, Precision::FP32},
+                               {TensorDescCreatorTypes::ncsp, Precision::I32}},
+                              {{TensorDescCreatorTypes::ncsp}});
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/topk.cpp:96]: We have to explicitly specify FP32 as output precision as the only precision supported by the implementation

File: inference-engine/src/mkldnn_plugin/nodes/topk.cpp [hunk 2]
```diff
@@ -20,51 +23,53 @@ namespace InferenceEngine {
...
-                    if (dst_dims[i] != dst_idx_dims[i])
-                        THROW_IE_EXCEPTION << layer->name << " Input/output tensor dimension mismatch";
-                }
+            auto topKOp = ngraph::as_type_ptr<const ngraph::op::v1::TopK>(op);
+            if (!topKOp) {
+                errorMessage = "Node is not an instance of the TopK operation.";
+                return false;
+            }
+            if (topKOp->get_mode() != ngraph::op::TopKMode::MAX &&
+                    topKOp->get_mode() != ngraph::op::TopKMode::MIN) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/topk.cpp:38]: please use https://github.com/dmitry-gorokhov/openvino/blob/master/inference-engine/src/mkldnn_plugin/utils/general_utils.h#L26

File: inference-engine/src/mkldnn_plugin/nodes/topk.cpp [hunk 3]
```diff
@@ -20,51 +23,53 @@ namespace InferenceEngine {
...
-                if (dst_dims.size() != dst_idx_dims.size())
-                    THROW_IE_EXCEPTION << layer->name << " Incorrect output tensor dimension sizes";
-
-                for (size_t i = 0; i < dst_dims.size(); i++) {
-                    if (dst_dims[i] != dst_idx_dims[i])
-                        THROW_IE_EXCEPTION << layer->name << " Input/output tensor dimension mismatch";
-                }
+            auto topKOp = ngraph::as_type_ptr<const ngraph::op::v1::TopK>(op);
+            if (!topKOp) {
+                errorMessage = "Node is not an instance of the TopK operation.";
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/topk.cpp:34]: Not fully correct wording. We have to specify the opset (e.g. `the v1::TopK operation`)

File: inference-engine/src/mkldnn_plugin/nodes/topk.cpp (hunk 1 above)
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/topk.cpp]: fixed

File: inference-engine/src/mkldnn_plugin/nodes/topk.cpp (hunk 2 above)
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/topk.cpp:39]: Fixed

File: inference-engine/src/mkldnn_plugin/nodes/topk.cpp (hunk 3 above)
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/topk.cpp]: FIxed

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**